Automattic/jetpack
# Add an API endpoint for performing bulk actions on feedback responses

## Proposed changes:

This patch adds an endpoint for asynchronously performing bulk actions on Jetpack Forms dashboard items. Currently supported actions are:

- `mark_as_spam`: Marks the item as spam.
- `mark_as_not_spam`: Removes the item from spam and marks it safe. Will also trigger re-sending of the notification email like the old admin page.
- `trash`: Moves the item to trash.
- `untrash`: Removes the item from trash.
- `delete`: Deletes the item and associated data completely.

In order to trigger an action, you'll need to send a `POST` request to `/wp-json/wpcom/v2/forms/responses` with an additional `bulk_action=` directive in the content type header and a list of `post_ids` as the payload. For example:

```
wp.apiFetch(  {
	path: '/wpcom/v2/forms/responses',
	method: 'POST',
	headers: {
		'Content-Type': 'application/json;bulk_action=mark_as_spam'
	},
	body: JSON.stringify( { post_ids: [ 64 ] } ),
}  );
```

The above code would mark post with ID `64` as spam. If the request is successful, the server will respond with code `200` and an empty object. In case of an error, there'll be an `error` message in the payload and the respective response code will be set.

### Other information:


## Jetpack product discussion



pcsBup-Pl-p2

## Does this pull request change what data or activity we track or use?



No.

## Testing instructions:






- Go to `/wp-admin/admin.php?page=jetpack-forms`, pick a feedback item to test with and get its ID.
- Proceed to manually make the request above with the given post ID.
- The feedback item should be moved to spam, you can verify this on the old WP Admin view if you need to.
- Repeat the above for all actions: `mark_as_spam`, `mark_as_not_spam`, `trash`, `untrash` and `delete`.
